# 05. Comparação do SOM contra Algoritmos Clássicos (Baselines)

Neste notebook comparamos o desempenho do SOM contra K-Means, PCA + K-Means, Agglomerative Clustering e DBSCAN.

In [1]:
import sys
import os
import pandas as pd
import numpy as np
import json
import matplotlib.pyplot as plt
from PIL import Image

# Garantir que a pasta src esteja no path
sys.path.append(os.path.abspath("../src"))
from load_data import load_synthetic_control, CLASS_NAMES

print("Ambiente importado com sucesso!")

Ambiente importado com sucesso!


### Tabela Completa de Métricas de Benchmarking
Carregamos e analisamos a comparação final contendo todos os baselines.

In [2]:
comparison_df = pd.read_csv("../outputs/metrics/model_comparison_results.csv")
comparison_df

,Modelo,ARI,NMI,Silhouette,Davies-Bouldin,Calinski-Harabasz,Pureza Neurônios,Erro Quantização,Erro Topográfico
0,SOM 5x5,0.658436,0.806309,0.154788,1.713133,177.523531,0.919301,4.734788,0.240000
1,SOM 7x7,0.609506,0.755073,0.187572,1.581400,204.789802,0.929878,4.244013,0.120000
2,SOM 10x10,0.633736,0.771309,0.204238,1.597494,208.105158,0.959387,3.903503,0.150000
3,SOM 12x12,0.631266,0.769203,0.200928,1.602735,208.384868,0.989958,3.751181,0.148333
4,SOM 15x15,0.621738,0.762536,0.204005,1.606272,209.654510,0.987449,3.555549,0.065000
5,SOM 20x20,0.587878,0.759125,0.262919,1.207376,213.273778,0.996732,3.334735,0.045000
6,K-Means (Direto),0.568100,0.741081,0.271040,1.210506,214.486932,NaN,NaN,NaN
7,PCA + K-Means,0.609621,0.718856,0.148107,1.950508,182.776244,NaN,NaN,NaN
8,Agglomerative (K=6),0.616043,0.800791,0.250994,1.202078,209.523729,NaN,NaN,NaN
9,DBSCAN,0.141834,0.401870,0.181896,2.349495,55.964029,NaN,NaN,NaN


### Análise de Desempenho:
1. **SOM 5x5** e **SOM 10x10** apresentam os maiores valores de ARI e NMI (~0.65 e ~0.80).
2. O **K-Means direto** e o **Agglomerative Clustering** alcançam boa coesão local (Silhouette de 0.27 e 0.25), mas erram mais o particionamento global em relação às classes reais (ARI de 0.56 e 0.61).
3. O **DBSCAN** obteve baixa performance (ARI de 0.14) devido à alta dimensionalidade das séries que dispersa o conceito de densidade.